components/brave_news/browser/feed_v2_builder.cc

```diff
@@ -1109,10 +1109,12 @@ void FeedV2Builder::GenerateFeed(
             feed->source_hash = builder->hash_;
 
             if (feed->items.empty()) {
-              if (builder->raw_feed_items_.size() == 0) {
-                feed->error = mojom::FeedV2Error::ConnectionError;
-              } else if (builder->subscribed_count_ == 0) {
+              if (builder->subscribed_count_ == 0 &&
```

Comment:
I'm curious if this meant to be equivalent to

```
if (builder->subscribed_count || !publishers_controller_->GetLastPublishers()
                       .empty())
```

If so, could `2 == true` case be incorrect, as it returns false?

Reply:
I think this is okay as-is. It's checking whether you're subscribed to anything which requires checking two things:
1. You must have a non-zero subscribed count
2. Publishers must have loaded (if not, there won't be any publishers/channels, so you won't be subscribed to anything).

I needed to swap the `ConnectionError` and `NoFeeds` cases around, because if you aren't subscribed to anything we won't fetch any feeds :laughing:

Reply:
Thanks for the explanation. I seem to misread the line for something like `builder->subscribed_count == !publishers_->GetlastPublishers().empty()`